kedro/framework/cli/jupyter.py

```diff
@@ -247,46 +238,6 @@ def convert_notebook(
     secho("Done!", color="green")  # type: ignore
 
 
-def _build_jupyter_command(  # pylint: disable=too-many-arguments
-    base: str,
-    ip_address: str,
-    all_kernels: bool,
-    args: Iterable[str],
-    idle_timeout: int,
-    project_name: str = "Kedro",
-) -> List[str]:
-    cmd = [
-        base,
-        "--ip",
-        ip_address,
-        f"--MappingKernelManager.cull_idle_timeout={idle_timeout}",
-        f"--MappingKernelManager.cull_interval={idle_timeout}",
-    ]
-
-    if not all_kernels:
-        kernel_name = re.sub(r"[^\w]+", "", project_name).strip() or "Kedro"
-
-        cmd += [
-            "--NotebookApp.kernel_spec_manager_class="
-            "kedro.framework.cli.jupyter.SingleKernelSpecManager",
-            f"--KernelSpecManager.default_kernel_name='{kernel_name}'",
-        ]
-
-    return cmd + list(args)
-
-
-def _build_jupyter_env(kedro_env: str) -> Dict[str, Any]:
-    """Build the environment dictionary that gets injected into the subprocess running
-    Jupyter. Since the subprocess has access only to the environment variables passed
-    in, we need to copy the current environment and add ``KEDRO_ENV``.
-    """
-    if not kedro_env:
-        return {}
-    jupyter_env = os.environ.copy()
-    jupyter_env["KEDRO_ENV"] = kedro_env
-    return {"env": jupyter_env}
-
```

Comment:
This is now done just as `os.environ["KEDRO_ENV"] = env`, the same way it's always been done in `kedro ipython`. I don't think the docstring here was right actually - `subprocess.run` will inherit the current environment's variables. And there's no need to worry about contaminating the environment, since adding something to `os.environ` is not persisted after the session is closed.

Reply:
Maybe it was true for older versions of Python, but it's still worth testing that KEDRO_ENV is definitely loading up the config from the correct environment.

Reply:
Yeah, I checked this already manually and all seemed to work ok. Good point about older versions of Python though  - I will double check with another version.